src/Access.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Storj\Uplink;
+
+use FFI;
+use FFI\CData;
+use Storj\Uplink\Exception\UplinkException;
+use Storj\Uplink\Internal\Scope;
+use Storj\Uplink\Internal\Util;
+
+/**
+ * Access grant
+ */
+class Access
+{
+    /**
+     * With libuplink.so and header files loaded
+     */
+    private FFI $ffi;
+
+    /**
+     * The associated C struct of type Access
+     */
+    private CData $cAccess;
+
+    /**
+     * To free memory when this object is done
+     */
+    private Scope $scope;
+
+    public function __construct(
+        FFI $ffi,
+        CData $cAccess,
+        Scope $scope
+    ) {
+        $this->ffi = $ffi;
+        $this->cAccess = $cAccess;
+        $this->scope = $scope;
+    }
+
+    /**
+     * @throws UplinkException
```

Comment:
To fill PHPDoc, it's best to consult the existing godoc at https://pkg.go.dev/storj.io/uplink?tab=doc

Reply:
Done